### PR TITLE
core(audit): align meta properties with LHR

### PR DESCRIPTION
**Pull request by patrickhulce**
We updated the output, but not the input. Before we get stuck with helpText for a year and officially ship the major, we should do that :)

name -> id
description -> title
helpText -> description

### CODE REVIEW

**Comment by paulirish** [comment]
there's two more instances of 'helptext' that should probably be renamed (though its not a bug)

**Comment by patrickhulce** [comment]
hm @paulirish where? vscode isn't finding any more non-string ones :/
image

image

File: lighthouse-core/audits/accessibility/th-has-data-cells.js [hunk 1]
```diff
@@ -18,12 +18,13 @@ class THHasDataCells extends AxeAudit {
...
-      name: 'th-has-data-cells',
-      description: '`<th>` elements and elements with `[role="columnheader"/"rowheader"]` have ' +
+      id: 'th-has-data-cells',
+      title: '`<th>` elements and elements with `[role="columnheader"/"rowheader"]` have ' +
           'data cells they describe.',
-      failureDescription: '`<th>` elements and elements with ' +
+      failureTitle: '`<th>` elements and elements with ' +
           '`[role="columnheader"/"rowheader"]` do not have data cells they describe.',
-      helpText: 'Screen readers have features to make navigating tables easier. Ensuring table ' +
+      // eslint-disable-next-line max-len
```
Comment (brendankenny) [inline lighthouse-core/audits/accessibility/th-has-data-cells.js]: other shifts were fixed, just fix this one as well? :)

File: lighthouse-core/audits/byte-efficiency/uses-text-compression.js [hunk 2]
```diff
@@ -21,10 +21,11 @@ class ResponsesAreCompressed extends ByteEfficiencyAudit {
...
   static get meta() {
     return {
-      name: 'uses-text-compression',
+      id: 'uses-text-compression',
       scoreDisplayMode: ByteEfficiencyAudit.SCORING_MODES.NUMERIC,
-      description: 'Enable text compression',
-      helpText: 'Text-based responses should be served with compression (gzip, deflate or brotli)' +
+      title: 'Enable text compression',
+      // eslint-disable-next-line max-len
+      description: 'Text-based responses should be served with compression (gzip, deflate or brotli)' +
```
Comment (brendankenny) [inline lighthouse-core/audits/byte-efficiency/uses-text-compression.js]: easy shift :)

File: lighthouse-core/audits/first-contentful-paint.js [hunk 3]
```diff
@@ -14,9 +14,10 @@ class FirstContentfulPaint extends Audit {
    */
   static get meta() {
     return {
-      name: 'first-contentful-paint',
-      description: 'First Contentful Paint',
-      helpText: 'First contentful paint marks the time at which the first text/image is painted. ' +
+      id: 'first-contentful-paint',
+      title: 'First Contentful Paint',
+      // eslint-disable-next-line max-len
+      description: 'First contentful paint marks the time at which the first text/image is painted. ' +
```
Comment (brendankenny) [inline lighthouse-core/audits/first-contentful-paint.js]: 😬😬🎉

File: lighthouse-core/audits/first-meaningful-paint.js [hunk 4]
```diff
@@ -14,9 +14,10 @@ class FirstMeaningfulPaint extends Audit {
    */
   static get meta() {
     return {
-      name: 'first-meaningful-paint',
-      description: 'First Meaningful Paint',
-      helpText: 'First Meaningful Paint measures when the primary content of a page is visible. ' +
+      id: 'first-meaningful-paint',
+      title: 'First Meaningful Paint',
+      // eslint-disable-next-line max-len
```
Comment (brendankenny) [inline lighthouse-core/audits/first-meaningful-paint.js]: 🌵 🍰 😰

File: lighthouse-core/audits/manual/pwa-page-transitions.js [hunk 5]
```diff
@@ -17,10 +17,11 @@ class PWAPageTransitions extends ManualAudit {
    */
   static get meta() {
     return Object.assign({
-      name: 'pwa-page-transitions',
-      helpText: 'Transitions should feel snappy as you tap around, even on a slow network, a key ' +
+      id: 'pwa-page-transitions',
+      // eslint-disable-next-line max-len
```
Comment (brendankenny) [inline lighthouse-core/audits/manual/pwa-page-transitions.js]: 🏷 🍃 🚈

File: lighthouse-core/audits/redirects-http.js [hunk 6]
```diff
@@ -13,10 +13,11 @@ class RedirectsHTTP extends Audit {
...
   static get meta() {
     return {
-      name: 'redirects-http',
-      description: 'Redirects HTTP traffic to HTTPS',
-      failureDescription: 'Does not redirect HTTP traffic to HTTPS',
-      helpText: 'If you\'ve already set up HTTPS, make sure that you redirect all HTTP traffic ' +
+      id: 'redirects-http',
+      title: 'Redirects HTTP traffic to HTTPS',
+      failureTitle: 'Does not redirect HTTP traffic to HTTPS',
+      // eslint-disable-next-line max-len
```
Comment (brendankenny) [inline lighthouse-core/audits/redirects-http.js]: ✂️ 🍴 🚚

File: lighthouse-core/audits/without-javascript.js [hunk 7]
```diff
@@ -13,10 +13,11 @@ class WithoutJavaScript extends Audit {
...
   static get meta() {
     return {
-      name: 'without-javascript',
-      description: 'Contains some content when JavaScript is not available',
-      failureDescription: 'Does not provide fallback content when JavaScript is not available',
-      helpText: 'Your app should display some content when JavaScript is disabled, even if it\'s ' +
+      id: 'without-javascript',
+      title: 'Contains some content when JavaScript is not available',
+      failureTitle: 'Does not provide fallback content when JavaScript is not available',
+      // eslint-disable-next-line max-len
```
Comment (brendankenny) [inline lighthouse-core/audits/without-javascript.js]: 👕 💃 ☄️

File: lighthouse-core/audits/works-offline.js [hunk 8]
```diff
@@ -14,10 +14,11 @@ class WorksOffline extends Audit {
...
   static get meta() {
     return {
-      name: 'works-offline',
-      description: 'Responds with a 200 when offline',
-      failureDescription: 'Does not respond with a 200 when offline',
-      helpText: 'If you\'re building a Progressive Web App, consider using a service worker so ' +
+      id: 'works-offline',
+      title: 'Responds with a 200 when offline',
+      failureTitle: 'Does not respond with a 200 when offline',
+      // eslint-disable-next-line max-len
```
Comment (brendankenny) [inline lighthouse-core/audits/works-offline.js]: 🏸 👇 👾

File: lighthouse-core/test/fixtures/invalid-audits/missing-generate-audit-result.js [hunk 9]
```diff
@@ -9,9 +9,9 @@
 class MissingRequiredArtifacts {
   static get meta() {
     return {
-      name: 'missing-required-artifacts',
-      description: 'Missing required artifacts',
-      helpText: 'This is missing required artifacts',
+      id: 'missing-required-artifacts',
```
Comment (brendankenny) [inline lighthouse-core/test/fixtures/invalid-audits/missing-generate-audit-result.js]: maybe we should update these (and name of class) while we're noticing it?

File: lighthouse-core/test/fixtures/invalid-audits/missing-description.js [hunk 10]
```diff
@@ -11,8 +11,9 @@ const LighthouseAudit = require('../../../audits/audit');
 class MissingRequiredArtifacts extends LighthouseAudit {
   static get meta() {
     return {
-      name: 'missing-description',
-      helpText: 'This is missing required description (and failure description)',
+      id: 'missing-description',
+      title: 'Missing required artifacts',
```
Comment (brendankenny) [inline lighthouse-core/test/fixtures/invalid-audits/missing-description.js]: description

File: lighthouse-core/test/fixtures/invalid-audits/missing-audit.js [hunk 11]
```diff
@@ -11,9 +11,9 @@ const LighthouseAudit = require('../../../audits/audit');
 class MissingRequiredArtifacts extends LighthouseAudit {
   static get meta() {
     return {
-      name: 'missing-category',
-      description: 'Missing required artifacts',
-      helpText: 'This is missing required artifacts',
+      id: 'missing-category',
```
Comment (brendankenny) [inline lighthouse-core/test/fixtures/invalid-audits/missing-audit.js]: we also got these wrong :)

File: docs/architecture.md [hunk 12]
```diff
@@ -15,9 +15,9 @@ _Some incomplete notes_
 
 ### Audit/Report terminology
 * **Category** - Roll-up collection of audits and audit groups into a user-facing section of the report (eg. `Best Practices`). Applies weighting and overall scoring to the section. Examples: PWA, Accessibility, Best Practices.
-* **Audit description** - Short user-visible title for the successful audit. eg. “All image elements have `[alt]` attributes.”
-* **Audit failure description** - Short user-visible title for a failing  audit. eg. “Some image elements do not have `[alt]` attributes.”
-* **Audit help text** - Explanation of why the user should care about the audit. Not necessarily how to fix it, unless there is no external link that explains it. ([See helpText guidelines](CONTRIBUTING.md#helptext-guidelines)). eg. “Informative elements should aim for short, descriptive alternate text. Decorative elements can be ignored with an empty alt attribute. [Learn more].”
+* **Audit title** - Short user-visible title for the successful audit. eg. “All image elements have `[alt]` attributes.”
+* **Audit failure title** - Short user-visible title for a failing  audit. eg. “Some image elements do not have `[alt]` attributes.”
```
Comment (brendankenny) [inline docs/architecture.md]: should we change it to `**Audit failureTitle**` just to be crystal clear it's that property name?

File: lighthouse-core/config/config.js [hunk 13]
```diff
@@ -8,13 +8,13 @@
 const defaultConfigPath = './default-config.js';
 const defaultConfig = require('./default-config.js');
 const fullConfig = require('./full-config.js');
-const constants = require('./constants');
+const constants = require('./constants.js');
```
Comment (brendankenny) [inline lighthouse-core/config/config.js]: :)

File: lighthouse-core/test/config/config-test.js [hunk 14]
```diff
@@ -8,7 +8,7 @@
 const Config = require('../../config/config');
 const assert = require('assert');
 const path = require('path');
-const defaultConfig = require('../../config/default-config.js');
+const defaultConfig = require('../../config/default-config');
```
Comment (brendankenny) [inline lighthouse-core/test/config/config-test.js]: :( :(

File: lighthouse-core/test/fixtures/invalid-audits/missing-description.js (hunk 10 above)
Comment (paulirish) [reply lighthouse-core/test/fixtures/invalid-audits/missing-description.js]: err.. if its the same test then we'd leave out `title` and `failureTitle` but provide a description.

File: lighthouse-core/test/config/config-test.js [hunk 15]
```diff
@@ -246,20 +246,20 @@ describe('Config', () => {
...
 
     assert.throws(_ => new Config({
-      audits: [basePath + '/missing-help-text'],
-    }), /meta.helpText property/);
+      audits: [basePath + '/missing-description'],
+    }), /meta.description property/);
 
     assert.throws(_ => new Config({
       audits: [
         class EmptyStringHelpTextAudit extends Audit {
```
Comment (paulirish) [inline lighthouse-core/test/config/config-test.js]: > hm @paulirish where? vscode isn't finding any more non-string ones :/

yeah i was just looking for any of them. relax that case sensitivity ;)

helptext -> description

File: lighthouse-core/test/config/config-test.js [hunk 16]
```diff
@@ -246,20 +246,20 @@ describe('Config', () => {
...
           static get meta() {
             return {
-              name: 'empty-string-help-text',
-              description: 'description',
-              helpText: '',
+              id: 'empty-string-help-text',
```
Comment (paulirish) [inline lighthouse-core/test/config/config-test.js]: helptext -> description

and you fixed the one in valid-custom-audit already. :)

File: lighthouse-core/test/config/config-test.js [hunk 17]
```diff
@@ -221,21 +221,21 @@ describe('Config', () => {
...
 
     assert.throws(_ => new Config({
-      audits: [basePath + '/missing-description'],
-    }), /meta.description property/);
+      audits: [basePath + '/missing-title'],
+    }), /meta.title property/);
 
     assert.throws(_ => new Config({
       audits: [
         class BinaryButNoFailureDescAudit extends Audit {
```
Comment (paulirish) [inline lighthouse-core/test/config/config-test.js]: failureTitle

File: lighthouse-core/test/fixtures/invalid-audits/missing-description.js (hunk 10 above)
Comment (brendankenny) [reply lighthouse-core/test/fixtures/invalid-audits/missing-description.js]: Ha, I meant that the title said it was missing artifacts but good point

File: lighthouse-core/test/fixtures/invalid-audits/missing-generate-audit-result.js (hunk 9 above)
Comment (patrickhulce) [reply lighthouse-core/test/fixtures/invalid-audits/missing-generate-audit-result.js]: done

File: docs/architecture.md (hunk 12 above)
Comment (patrickhulce) [reply docs/architecture.md]: done

File: lighthouse-core/test/config/config-test.js (hunk 14 above)
Comment (patrickhulce) [reply lighthouse-core/test/config/config-test.js]: oooooh wait I think it's trying to match the ones around it! so clever :D

File: lighthouse-core/test/fixtures/invalid-audits/missing-description.js (hunk 10 above)
Comment (patrickhulce) [reply lighthouse-core/test/fixtures/invalid-audits/missing-description.js]: we already have `missing-title` for that, this one is really the help text check, git diff is confused

File: lighthouse-core/test/config/config-test.js (hunk 16 above)
Comment (patrickhulce) [reply lighthouse-core/test/config/config-test.js]: oh gotcha :) fixed!

File: lighthouse-core/test/config/config-test.js (hunk 15 above)
Comment (patrickhulce) [reply lighthouse-core/test/config/config-test.js]: fixed

File: lighthouse-core/test/config/config-test.js (hunk 17 above)
Comment (patrickhulce) [reply lighthouse-core/test/config/config-test.js]: done

**Pull request merged**